src/js/oncoprintruleset.js

```diff
@@ -935,11 +941,69 @@ var Rule = (function () {
     return Rule;
 })();
 
+var GradientCategoricalRuleSet = (function() {
+
+        function GradientCategoricalRuleSet(params) {
+	        RuleSet.call(this, params);
+	        this.prototype = Object.create(RuleSet.prototype);
+                // For the GradientCategoricalRuleSet a datum must always have a 
+                // value (for heatmap clustering) and can have a category attribute. 
+                // A datum is 'NA' when not meeting the requirements for the GradientRuleSet.
+                // To achieve correct evaluation, the CategoricalRuleSet is asked not to
+                // contribute an `NA` rule.
```

Comment:
good idea, and good comment

Reply:
except, I don't think there's a need to couple clustering with this ruleset. clustering is a separate functionality. so let's just say that, arbitrarily, there should always be a value. and please remove that from the comment on line 993 as well.

Reply:
I updated the comments to remove references to clustering.